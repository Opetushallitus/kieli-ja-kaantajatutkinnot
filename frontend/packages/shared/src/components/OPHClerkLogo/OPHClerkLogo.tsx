import { FC } from 'react';

import OPHLogo from '../../statics/svg/oph_logo_textless.svg';
import { Svg } from '../Svg/Svg';
import { Text } from '../Text/Text';
import './OPHClerkLogo.scss';

interface OPHClerkLogoProps {
  className?: string;
  mainLabel: string;
  subLabel: string;
  alt: string;
}

export const OPHClerkLogo: FC<OPHClerkLogoProps> = ({
  className,
  mainLabel,
  subLabel,
  alt,
}) => {
  return (
    <div className={`oph-clerk-logo columns ${className}`}>
      <Svg className="oph-clerk-logo__img" src={OPHLogo} alt={alt} />
      <div className="rows">
        <Text className="oph-clerk-logo__label--bold">{mainLabel}</Text>
        <Text className="oph-clerk-logo__label">{subLabel}</Text>
      </div>
    </div>
  );
};
