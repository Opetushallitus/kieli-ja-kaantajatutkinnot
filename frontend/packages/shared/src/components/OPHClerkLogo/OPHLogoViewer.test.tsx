import renderer from 'react-test-renderer';

import { OPHClerkLogo } from './OPHClerkLogo';

describe('OPHClerkLogo', () => {
  it('should render OPHClerkLogo correctly', () => {
    const tree = renderer
      .create(
        <OPHClerkLogo mainLabel="MainLabel" subLabel="SubLabel" alt="ImgAlt" />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
