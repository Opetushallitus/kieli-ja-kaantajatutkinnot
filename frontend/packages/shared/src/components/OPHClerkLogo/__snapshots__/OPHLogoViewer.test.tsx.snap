// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OPHClerkLogo should render OPHClerkLogo correctly 1`] = `
<div
  className="oph-clerk-logo columns undefined"
>
  <img
    alt="ImgAlt"
    className="oph-clerk-logo__img"
    src="test"
  />
  <div
    className="rows"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 oph-clerk-logo__label--bold css-ahj2mt-MuiTypography-root"
    >
      MainLabel
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 oph-clerk-logo__label css-ahj2mt-MuiTypography-root"
    >
      SubLabel
    </p>
  </div>
</div>
`;
